specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2021-01-01/kusto.json

```diff
@@ -0,0 +1,4925 @@
+        },
+        "hotCachePeriod": {
+          "type": "string",
+          "format": "duration",
+          "description": "The time the data should be kept in cache for fast queries in TimeSpan."
+        },
+        "statistics": {
+          "$ref": "#/definitions/DatabaseStatistics",
+          "description": "The statistics of the database."
+        },
+        "leaderClusterResourceId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The name of the leader cluster"
+        },
+        "attachedDatabaseConfigurationName": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The name of the attached database configuration cluster"
+        },
+        "principalsModificationKind": {
+          "type": "string",
+          "readOnly": true,
+          "enum": [
+            "Union",
+            "Replace",
+            "None"
+          ],
+          "x-ms-enum": {
+            "name": "PrincipalsModificationKind",
+            "modelAsString": true
+          },
+          "description": "The principals modification kind of the database"
+        }
+      },
+      "description": "Class representing the Kusto database properties."
+    },
+    "EventHubConnectionProperties": {
+      "properties": {
+        "eventHubResourceId": {
+          "type": "string",
+          "description": "The resource ID of the event hub to be used to create a data connection."
+        },
+        "consumerGroup": {
+          "type": "string",
+          "description": "The event hub consumer group."
+        },
+        "tableName": {
+          "type": "string",
+          "description": "The table where the data should be ingested. Optionally the table information can be added to each message."
+        },
+        "mappingRuleName": {
+          "type": "string",
+          "description": "The mapping rule to be used to ingest the data. Optionally the mapping information can be added to each message."
+        },
+        "dataFormat": {
+          "$ref": "#/definitions/EventHubDataFormat",
+          "description": "The data format of the message. Optionally the data format can be added to each message."
+        },
+        "eventSystemProperties": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "System properties of the event hub"
+        },
+        "compression": {
+          "$ref": "#/definitions/Compression",
+          "description": "The event hub messages compression type"
+        },
+        "provisioningState": {
+          "$ref": "#/definitions/ProvisioningState",
+          "description": "The provisioned state of the resource."
+        },
+        "managedIdentityResourceId": {
+          "type": "string",
+          "description": "The resource ID of a managed identity (system or user assigned) to be used to authenticate with event hub."
```

Comment:
What resource does the managed identity live on?  What checks do you do to ensure the resource id provided here should be allowed?  How does someone specify a system assigned identity using this property?

Reply:
The managed identity can be either the system assigned identity that is enabled on the cluster resource (the value would be the resource Id of the cluster). Or it can be a managed identity that is pre assigned to the cluster resource.
Our RP service validates:
a) The value is a well formatted resource id. 
b) The value is assigned to the cluster (either the system assigned identity of the cluster which was enabled in advance or a managed identity that was assigned to it in advance).